gateway/webhook/webhook.go

```diff
@@ -289,39 +290,38 @@ func (bt *batchWebhookTransformerT) batchTransformLoop() {
 		var payloadArr [][]byte
 		var webRequests []*webhookT
 		for _, req := range breq.batchRequest {
-			body, err := io.ReadAll(req.request.Body)
-			_ = req.request.Body.Close()
+			var body []byte
+			if req.request.Body != nil {
+				body, err = io.ReadAll(req.request.Body)
+				_ = req.request.Body.Close()
 
-			if err != nil {
-				req.done <- transformerResponse{Err: response.GetStatus(response.RequestBodyReadFailed)}
-				continue
-			}
-
-			if slices.Contains(bt.webhook.config.sourceListForParsingParams, strings.ToLower(breq.sourceType)) {
-				queryParams := req.request.URL.Query()
-				paramsBytes, err := json.Marshal(queryParams)
 				if err != nil {
-					req.done <- transformerResponse{Err: response.GetStatus(response.ErrorInMarshal)}
+					req.done <- transformerResponse{Err: response.GetStatus(response.RequestBodyReadFailed)}
 					continue
 				}
+			}
 
-				closingBraceIdx := bytes.LastIndexByte(body, '}')
-				if closingBraceIdx == -1 {
+			jsonBody := "{}" // Default to an empty JSON object if no body is present
+			if len(body) > 0 {
+				if !json.Valid(body) {
 					req.done <- transformerResponse{Err: response.GetStatus(response.InvalidJSON)}
 					continue
 				}
-				appendData := []byte(`, "query_parameters": `)
-				appendData = append(appendData, paramsBytes...)
-				body = append(body[:closingBraceIdx], appendData...)
-				body = append(body, '}')
+				jsonBody = string(body)
 			}
 
-			if !json.Valid(body) {
-				req.done <- transformerResponse{Err: response.GetStatus(response.InvalidJSON)}
-				continue
+			if slices.Contains(bt.webhook.config.sourceListForParsingParams, strings.ToLower(breq.sourceType)) {
+				queryParams := req.request.URL.Query()
+				for key, values := range queryParams {
```

Comment:
you are breaking existing contract, currently query_parameters is used as key

Reply:
corrected